crates/subspace-service/src/sync_from_dsn/segment_header_downloader.rs

```diff
@@ -5,18 +5,27 @@ use std::collections::{BTreeSet, HashMap};
 use std::error::Error;
 use std::pin::pin;
 use std::sync::Arc;
+use std::time::Duration;
 use subspace_core_primitives::segments::{SegmentHeader, SegmentIndex};
 use subspace_networking::libp2p::PeerId;
 use subspace_networking::protocols::request_response::handlers::segment_header::{
     SegmentHeaderRequest, SegmentHeaderResponse,
 };
 use subspace_networking::Node;
-use tracing::{debug, error, trace, warn};
+use tokio::time::sleep;
+use tracing::{debug, error, info, trace, warn};
 
 const SEGMENT_HEADER_NUMBER_PER_REQUEST: u64 = 1000;
 /// Initial number of peers to query for segment header
 const SEGMENT_HEADER_CONSENSUS_INITIAL_NODES: usize = 20;
 
+/// Number of retries for the last segment header peer checks, and the segment header batch
+/// request. Each new segment index range resets the retry count.
+const SEGMENT_HEADER_RETRIES: u32 = 5;
+
+/// Time between retries for segment headers.
+const SEGMENT_HEADER_RETRY_DELAY: Duration = Duration::from_secs(30);
```

Comment:
Why is this delay needed at all? It'll just make everything super slow, especially right after first iteration to potentially already disconnected peer. Is there a value provided by it? I'd remove it and try random peers asap.